pkg/library/home/persistentHome.go

```diff
@@ -95,3 +115,32 @@ func storageStrategySupportsPersistentHome(workspace *common.DevWorkspaceWithCon
 	storageClass := workspace.Spec.Template.Attributes.GetString(constants.DevWorkspaceStorageTypeAttribute, nil)
 	return storageClass != constants.EphemeralStorageClassType
 }
+
+// Add an init container that runs /entrypoint.sh to initialize the persistent home directory.
+// This function assumes that the provided image contains the /entrypoint.sh script.
+// If the script is not present in the image, the init container will not fail the workspace from starting.
+func addPersistentStorageInitContainer(dwTemplateSpecCopy *v1alpha2.DevWorkspaceTemplateSpec, image string) {
+	initContainerComponent := v1alpha2.Component{
+		Name: "init-persistent-home",
+		ComponentUnion: v1alpha2.ComponentUnion{
+			Container: &v1alpha2.ContainerComponent{
+				Container: v1alpha2.Container{
+					Image:   image,
+					Command: []string{"/bin/sh", "-c", "/entrypoint.sh || true"},
```

Comment:
Making this assumption creates coupling between DWO and the UDI

Reply:
@AObuchow it could be possible to introduce the initcontainer in the dashboard, I am interested in investigating